erdpy/proxy/core.py

```diff
@@ -120,3 +121,19 @@ def get_hyperblock(self, key) -> Any:
         response = do_get(url)
         response = response.get("hyperblock", {})
         return response
+
+    def send_transaction_and_wait_for_result(self, payload: Any) -> str:
+        url = f"{self.url}/transaction/send"
+        response = do_post(url, payload)
+        tx_hash = response.get("txHash")
+
+        num_runs = 100
+        for _ in range(0, num_runs):
+            last_nonce = self.get_last_block_nonce("metachain")
+            last_hyperblock = self.get_hyperblock(last_nonce)
+            finalized_transactions = last_hyperblock["transactions"]
+            for transaction in finalized_transactions:
+                if transaction["hash"] == tx_hash:
+                    return self.get_transaction(tx_hash=tx_hash, with_results=True)
+            time.sleep(1)
+        return ""
```

Comment:
```suggestion
        for _ in range(0, num_runs):
            if self.is_transaction_finalized(tx_hash):
                return self.get_transaction(tx_hash=tx_hash, with_results=True)
            time.sleep(1)
        return ""
        
   def is_transaction_finalized(self, tx_hash):
        last_nonce = self.get_last_block_nonce("metachain")
        last_hyperblock = self.get_hyperblock(last_nonce)
        finalized_transactions = last_hyperblock["transactions"]
        
        for transaction in finalized_transactions:
                if transaction["hash"] == tx_hash:
                    return True
                    
        return False
```

Reply:
Refactored. This will be called from another repo, PR is on the way.